codegen/testserver/generated_test.go

```diff
@@ -276,67 +462,109 @@ func TestGeneratedServer(t *testing.T) {
 		}, tracerLog)
 	})
 
-	t.Run("subscriptions", func(t *testing.T) {
-		t.Run("wont leak goroutines", func(t *testing.T) {
-			initialGoroutineCount := runtime.NumGoroutine()
-
-			sub := c.Websocket(`subscription { updated }`)
-
-			resolvers.tick <- "message"
+	t.Run("called in the correct order", func(t *testing.T) {
+		resolvers := &testResolver{tick: make(chan string, 1)}
+
+		configurableTracer := &configurableTracer{}
+
+		srv := httptest.NewServer(
+			handler.GraphQL(
+				NewExecutableSchema(Config{Resolvers: resolvers}),
+				handler.Tracer(configurableTracer),
+			))
+		defer srv.Close()
+		c := client.New(srv.URL)
+
+		steps := []string{
+			"StartOperationParsing",
+			"StartOperationValidation",
+			"StartOperationExecution",
+			"StartFieldExecution",
+			"StartFieldResolverExecution",
+			"StartFieldChildExecution",
+		}
 
-			var msg struct {
-				resp struct {
-					Updated string
+		assertStep := func(target string) func(ctx context.Context) {
```

Comment:
Are the arrays required in the configurable tracer? this would a bit clearer with something like:
```go
configurableTracer := &configurableTracer{
    StartOperationParsingFuncs: func(ctx context.Context) context.Context {
        return context.WithValue("StartOperationParsing", true)
    },
    StartOperationValidation: func(ctx context.Context) context.Context {
        assert.True(t, ctx.Value("StartOperationParsing")
        return context.WithValue("StartOperationValidation", true)
    },
    // etc
}
```

no need for anonymous helper funcs or appending to arrays?